airbyte-workers/src/main/java/io/airbyte/workers/DefaultReplicationWorker.java

```diff
@@ -29,29 +29,47 @@
 import org.slf4j.LoggerFactory;
 import org.slf4j.MDC;
 
+/**
+ * This worker is the "data shovel" of ETL. It is responsible for moving data from the Source
+ * container to the Destination container. It manages the full lifecycle of this process. This
+ * includes:
+ * <ul>
+ * <li>Starting the Source and Destination containers</li>
+ * <li>Passing data from Source to Destination</li>
+ * <li>Executing any configured map-only operations (Mappers) in between the Source and
+ * Destination</li>
+ * <li>Collecting metadata about the data that is passing from Source to Destination</li>
+ * <li>Listening for state messages emitted from the Destination to keep track of what data has been
+ * replicated.</li>
+ * <li>Handling shutdown of the Source and Destination</li>
+ * <li>Handling failure cases and attempting persist state for partially completed replications (so
+ * that the next replication can pick up where it left off instead of starting from the
+ * beginning)</li>
```

Comment:
This bullet says that this class handles persisting the state for partial completions, but does it also contain the logic to read this persistence and pick up from where it left off? Or does that happen somewhere else?

Reply:
good call out. adjusting it to say returning state